src/test/resources/hudson/plugins/ec2/WindowsData.yml

```diff
@@ -9,7 +9,7 @@ jenkins:
           - description:
             ami: "ami-abc123"
             labelString: "windows vs2019"
-            type: "M5aLarge"
+            type: "M5_A_LARGE"
```

Comment:
@jetersen @timja Any suggestions on how to retain compatibility with the old enumeration names for CasC purposes? This is an interesting case because the new `software.amazon.awssdk.services.ec2.model.InstanceType` is an enumeration, just like the old `com.amazonaws.services.ec2.model.InstanceType`, just using upper snake case rather than Pascal case.

Reply:
Not really other than that you map old to new yourself and potentially a new field name and deprecate the old one.

Reply:
I'm still working on it, but what I'm trying to achieve on my side is to accept strings and then resolve from the text value of the Enum.
Both enums in the old and new SDKs resolve to the actual API name
e.g. `t2.large` is `T2_LARGE("t2.large"),` in the new one and `T2Large("t2.large"),` in the old one.
We would then accept either `T2Large` (legacy) or `t2.large`.
This has the benefit of removing the implementation detail leaking into the choices in the UI while allowing the old configs to still be loaded

Reply:
That should work fine, best to use the name that users see in the UI / AWS documentation. If that's `t2.large` then great, (and supporting the old format)

Reply:
I also worry that we might need to do the same thing for data serialized to XML via XStream.

Reply:
I'm not too sure how that works, if it uses the same `@DataBoundConstructor`, my fix could work.

Here are my changes to allow loading with the old and new instance type string: https://github.com/coveord/ec2-plugin/tree/fix/JENKINS-73640 (I made the branch on top of yours, so it should merge cleanly)

Reply:
Just tested:
- Checked out master
- Ran a test instance
- Added a configuration with instance type `T3Micro`
- Stopped the test instance
- Checked out my branch
- Ran a test instance
Unfortunately, the instance type is now the first one in the list `a1.2xlarge`. Will try to figure out why

Reply:
Fixed in [my branch](https://github.com/coveord/ec2-plugin/tree/fix/JENKINS-73640), it was only a UI thing

So the test goes:
- Checked out master
- Ran a test instance
- Added a configuration with instance type T3Micro
  - In `work/config.xml`, the instance type is saved as `T3Micro`
- Stopped the test instance
- Checked out my branch
- Ran a test instance
- Go to the same cloud configuration
  - Now, `t3.micro` is selected
- Save the new configuration
  - In `work/config.xml`, the instance type is now saved as `t3.micro`
- Reload the cloud configuration page
  - `t3.micro` is still selected

Reply:
Thanks @dblanchette, I added a variation of your code to my branch. Are you interested in testing the latest incremental build with a real EC2 account?

Reply:
I have a former production instance on which I can install it, but I'm not sure when I'll have time. Thanks for integrating my changes!